File: backend/schemas/empresa.py
"""
Schemas Pydantic para el módulo de empresas.
Patrón: EmpresaBase → EmpresaCreate → EmpresaUpdate → EmpresaResponse → EmpresaListResponse
"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class EmpresaBase(BaseModel):
    nombre: str
    razon_social: Optional[str] = None
    cuit: Optional[str] = None
    direccion: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    logo_url: Optional[str] = None
    activa: bool = True


class EmpresaCreate(EmpresaBase):
    pass


class EmpresaUpdate(BaseModel):
    nombre: Optional[str] = None
    razon_social: Optional[str] = None
    cuit: Optional[str] = None
    direccion: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    logo_url: Optional[str] = None
    activa: Optional[bool] = None


class EmpresaActivaToggle(BaseModel):
    activa: bool


class EmpresaResponse(BaseModel):
    id: str
    nombre: str
    razon_social: Optional[str] = None
    cuit: Optional[str] = None
    direccion: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    logo_url: Optional[str] = None
    activa: bool
    created_at: datetime
    updated_at: Optional[datetime] = None


class EmpresaListResponse(BaseModel):
    items: list[EmpresaResponse]
    total: int

File: frontend/types/empresa.ts
export interface Empresa {
  id: string
  nombre: string
  razon_social: string | null
  cuit: string | null
  direccion: string | null
  telefono: string | null
  email: string | null
  logo_url: string | null
  activa: boolean
  created_at: string
  updated_at: string | null
}

export interface EmpresaCreate {
  nombre: string
  razon_social?: string
  cuit?: string
  direccion?: string
  telefono?: string
  email?: string
  logo_url?: string
  activa?: boolean
}

export interface EmpresaUpdate {
  nombre?: string
  razon_social?: string
  cuit?: string
  direccion?: string
  telefono?: string
  email?: string
  logo_url?: string
  activa?: boolean
}

export interface EmpresaListResponse {
  items: Empresa[]
  total: number
}
